### PR TITLE
Add unit tests covering Data.init<T>(buffer: Unsafe{Mutable}BufferPointer<T>)

### PR DESCRIPTION
**Pull request by itaiferber**
**What's in this pull request?**
Adds unit tests which exercise the changes in #22521.
(Addresses rdar://problem/47978737)

### CODE REVIEW

**Comment by itaiferber** [comment]
@swift-ci Please smoke test

File: test/stdlib/TestData.swift [hunk 1]
```diff
@@ -238,6 +238,68 @@ class TestData : TestDataSuper {
...
         expectEqual(2, data3.count)
     }
+
+    func testInitializationWithBufferPointer() {
+        let nilBuffer = UnsafeBufferPointer<UInt8>(start: nil, count: 0)
+        let data = Data(buffer: nilBuffer)
+        expectEqual(data, Data())
+
+        let validPointer = UnsafeMutablePointer<UInt8>.allocate(capacity: 2)
+        validPointer[0] = 0xCA;
```
Comment (itaiferber) [inline test/stdlib/TestData.swift]: Don't know why these semicolons snuck in here. Will remove

**Comment by itaiferber** [comment]
@swift-ci Please smoke test

**Review by phausler — APPROVED** [review]

**Review by atrick — APPROVED** [review]

**Comment by itaiferber** [comment]
@swift-ci Please smoke test

**Pull request merged**